ksql-functional-tests/src/test/resources/query-validation-tests/schemas.json

```diff
@@ -115,33 +143,47 @@
           "schema": {"name": "blah", "type": "record", "fields": [{"name": "FOO", "type": {"type": "map", "values": "int"}}]},
           "format": "AVRO"
         }
-      ],
-      "inputs": [
+       ],
+       "inputs": [
         {"topic": "input_topic", "value": {"a": 1, "b": 2, "c": 3}}
-      ],
-      "outputs": [
+       ],
+       "outputs": [
         {"topic": "OUTPUT", "value": {"FOO": {"a": 1, "b": 2, "c": 3}}}
-      ],
-      "post": {
+       ],
+       "post": {
         "sources": [
           {
             "name": "INPUT",
             "type": "stream",
             "valueSchema": "STRUCT<ROWTIME BIGINT, ROWKEY STRING, FOO MAP<STRING, INT>>"
           }
         ]
-      }
+       }
+    },
+    {
+       "name": "top level map - value - JSON",
+       "statements": [
+          "CREATE STREAM INPUT (foo MAP<STRING, INT>) WITH (kafka_topic='input_topic', value_format='JSON');",
```

Comment:
maybe also add a test for the example with ambiguity?

Reply:
Will do in follow up PR